src/Components/Bookshelf/index.js

```diff
@@ -2,11 +2,14 @@ import React, { useEffect, useState } from "react"
 import "./bookshelf.css"
 import BookCard from "../BookCard"
 
-function Bookshelf() {
+function Bookshelf({ claimed }) {
     const [bookCollection, setBookCollection] = useState([])
 
     useEffect(() => {
```

Comment:
This useEffect only works when you first load the page. If I use the nav links to switch between pages, it doesn't re-fetch the correct list of books. But it does work if I click one of the nav links, then refresh the page.

You need to pass claimed into the [] of useEffect to re-run the fetch when the links are clicked.